Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCheckPointFirewall.yaml

```diff
@@ -0,0 +1,260 @@
+      , "96","SCC-SP"
+      , "97","ETHERIP"
+      , "98","ENCAP"
+      , "99",""
+      , "100","GMTP"
+      , "101","IFMP"
+      , "102","PNNI"
+      , "103","PIM"
+      , "104","ARIS"
+      , "105","SCPS"
+      , "106","QNX"
+      , "107","A/N"
+      , "108","IPComp"
+      , "109","SNP"
+      , "110","Compaq-Peer"
+      , "111","IPX-in-IP"
+      , "112","VRRP"
+      , "113","PGM"
+      , "114",""
+      , "115","L2TP"
+      , "116","DDX"
+      , "117","IATP"
+      , "118","STP"
+      , "119","SRP"
+      , "120","UTI"
+      , "121","SMP"
+      , "122","SM (deprecated)"
+      , "123","PTP"
+      , "124","ISIS over IPv4"
+      , "125","FIRE"
+      , "126","CRTP"
+      , "127","CRUDP"
+      , "128","SSCOPMCE"
+      , "129","IPLT"
+      , "130","SPS"
+      , "131","PIPE"
+      , "132","SCTP"
+      , "133","FC"
+      , "134","RSVP-E2E-IGNORE"
+      , "135","Mobility Header"
+      , "136","UDPLite"
+      , "137","MPLS-in-IP"
+      , "138","manet"
+      , "139","HIP"
+      , "140","Shim6"
+      , "141","WESP"
+      , "142","ROHC"
+      , "143","Ethernet"
+      , "253",""
+      , "254",""
+      , "255","Reserved"];
+  let DirectionLookup=datatable(CommunicationDirection:string,NetworkDirection:string)
+  [
+      "Incoming","Inbound", 
+      "Outgoing","Outbound", 
+      "Internal","Local"];
+  let ActionLookup=datatable(DvcOriginalAction:string,DvcAction:string)
+  [
+      "Accept","Allow", 
+      "Allow","Allow", 
+      "Drop","Drop", 
+      "Reject","Deny", 
+      "Encrypt","Encrypt", 
+      "Decrypt","Decrypt",
+      "Bypass","Allow", 
+      "Block","Deny", 
+      "Detect","Deny", 
+      "Do not send","Drop",  
+      "Authorize","", 
+      "Deauthorize","", 
+      "Ask User","", 
+      "Extract",""];
+let NWParser=(disabled:bool=false){
+    CommonSecurityLog
+    | where not(disabled)
+    | where DeviceVendor=="Check Point" and DeviceProduct=="VPN-1 & FireWall-1"
+    | extend 
+            EventProduct = "Firewall",
+            EventCount = toint(1),
+            EventType = "NetworkSession",
+            EventSchema = "NetworkSession",
+            EventSchemaVersion = "0.2.4"
+    | parse AdditionalExtensions with * "rule_action=" DeviceAction ";" * "rule_uid=" NetworkRuleName ";" *
+    | parse AdditionalExtensions with * "loguid=" EventOriginalUid ";" * "origin=" Dvc ";" * "originsicname=CN\\=" DvcHostname "," *
+    | parse AdditionalExtensions with * "inzone=" SrcZone ";" * "outzone=" DstZone ";" * 
+    | parse AdditionalExtensions with * "conn_direction=" CommunicationDirection ";" 
+    | parse AdditionalExtensions with * "alert=" AlertName ";" *
+    | parse AdditionalExtensions with * "inspection_category=" InspectionCategory ";" * "inspection_item=" ThreatName ";"                   
+    | extend
+            ThreatCategory = coalesce(AlertName, InspectionCategory),
+            EventResult = case(
+                                isempty(DeviceAction), "", 
```

Comment:
EventResult cannot be empty. It can be NA. Also, many of the new DeviceActions you added are also Success. The best way is to add a column to the DeviceAction lookup with the matching EventResult.

Reply:
Lookup update and "NA" is included as well.